tests/unit/amo/components/TestAutoSearchInput.js

```diff
@@ -0,0 +1,580 @@
+      const root = render({ query: 'panda themes' });
+
+      const suggestion = createInternalSuggestion(
+        createFakeAutocompleteResult()
+      );
+      selectSuggestion({ root, suggestion });
+
+      const autosuggest = root.find(Autosuggest);
+      expect(autosuggest.prop('inputProps')).toMatchObject({
+        // Make sure the query was reset.
+        value: '',
+      });
+    });
+  });
+
+  describe('suggestion result', () => {
+    const renderSuggestion = ({
+      root = render(),
+      suggestionData = createInternalSuggestion(
+        createFakeAutocompleteResult()
+      ),
+    } = {}) => {
+      const _renderSuggestion = simulateAutosuggestCallback({
+        root, propName: 'renderSuggestion',
+      });
+      const suggestion = _renderSuggestion(suggestionData);
+      return shallow(<div>{suggestion}</div>).find(SearchSuggestion);
+    };
+
+    it('converts a suggestion result into a value', () => {
+      const name = 'uBlock Origin';
+      const suggestion = createInternalSuggestion(
+        createFakeAutocompleteResult({ name })
+      );
+
+      const root = render();
+      const getSuggestionValue = simulateAutosuggestCallback({
+        root, propName: 'getSuggestionValue',
+      });
+      const value = getSuggestionValue(suggestion);
+
+      expect(value).toEqual(name);
+    });
+
+    it('renders a suggestion', () => {
+      const name = 'uBlock Origin';
+      const suggestionData = createInternalSuggestion(
+        createFakeAutocompleteResult({ name })
+      );
+      const suggestion = renderSuggestion({ suggestionData });
+
+      expect(suggestion).toHaveProp('name', name);
+      expect(suggestion).toHaveProp('iconUrl', suggestionData.iconUrl);
+    });
+
+    it('renders a placeholder while loading', () => {
+      const { store } = dispatchClientMetadata();
+      store.dispatch(autocompleteStart({
+        errorHandlerId: 'some-error-handler',
+        filters: { query: 'ad blockers' },
+      }));
+
+      const root = render({ store });
+      const suggestionData = createInternalSuggestion(
+        createFakeAutocompleteResult()
+      );
+      const suggestion = renderSuggestion({ root, suggestionData });
+
+      expect(suggestion).toHaveProp('loading', true);
+    });
+
+    it('renders custom arrow alt text', () => {
+      const selectSuggestionText = 'Visit extension detail page';
+      const root = render({ selectSuggestionText });
+      const suggestion = renderSuggestion({ root });
+
+      expect(suggestion).toHaveProp('arrowAlt', selectSuggestionText);
+    });
+
+    it('renders default arrow alt text', () => {
+      const suggestion = renderSuggestion();
+
+      expect(suggestion).toHaveProp('arrowAlt', 'Go to the add-on page');
+    });
+  });
+
+  describe('getting suggestion data', () => {
+    const getSuggestions = (root) => {
+      const autosuggest = root.find(Autosuggest);
+      expect(autosuggest).toHaveProp('suggestions');
```

Comment:
I think it would make a nicer error when the tests fail. Without it, you'd see something like `undefined doesn't have X` when you try to work with a `suggestions` prop.